extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -344,6 +435,38 @@ export abstract class AzureAuth implements vscode.Disposable {
 	}
 
 
+	public async getTenantsMsal(token: string): Promise<Tenant[]> {
+		interface TenantResponse { // https://docs.microsoft.com/en-us/rest/api/resources/tenants/list
+			id: string
+			tenantId: string
+			displayName?: string
+			tenantCategory?: string
+		}
+		const tenantUri = url.resolve(this.metadata.settings.armResource.endpoint, 'tenants?api-version=2019-11-01');
+		try {
+			Logger.verbose('Fetching tenants', tenantUri);
+			const tenantResponse = await this.makeGetRequest(tenantUri, token);
+			const tenants: Tenant[] = tenantResponse.data.value.map((tenantInfo: TenantResponse) => {
+				Logger.verbose(`Tenant: ${tenantInfo.displayName}`);
+				return {
+					id: tenantInfo.tenantId,
+					displayName: tenantInfo.displayName ? tenantInfo.displayName : localize('azureWorkAccountDisplayName', "Work or school account"),
+					userId: token,
+					tenantCategory: tenantInfo.tenantCategory
+				} as Tenant;
+			});
+			const homeTenantIndex = tenants.findIndex(tenant => tenant.tenantCategory === 'Home');
+			if (homeTenantIndex >= 0) {
```

Comment:
It's useful for logic like this to explain what this is supposed to be doing in a comment since it may not be immediately clear to people reading this.